src/DocumentDB.ChangeFeedProcessor/PartitionControllerHealthnessEvaluator.cs

```diff
@@ -0,0 +1,60 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+//----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
+
+    internal class PartitionControllerHealthnessEvaluator : IPartitionController
+    {
+        private static readonly long UninitializedTicks = DateTime.MaxValue.Ticks;
+        private readonly IPartitionController inner;
+        private readonly long maxAllowedUnhealthyDuration;
+        private readonly IUnhealthyHandlingStrategy unhealthyStrategy;
+        private long firstUnhealthyTick = UninitializedTicks;
```

Comment:
why use ticks and not Timespan? With Timespans IMO it would be more natural and we won't need to use "checked".

Reply:
the reason is that I need to handle it in multithreaded env and I'd like to have it lock-less. So I used Interlock methods. They dont work with DateTime so I had to fallback to Ticks. Otherwise I totally agree with you